### ISSUE TITLE
Popup with child does not show tooltip when hovered

### ISSUE THREAD

**Issue by Roflraging** [ocornut/imgui#197]
Hey Omar,

I've been checking out some of the popup functionality you implemented in your last release and I noticed an issue in my application while using it with a child region inside the popup.  This code illustrates the issue:

```
        ImGui::Begin("Popup tooltip bug");

        static bool popup = false;
        popup ^= ImGui::Button("Popup");

        static bool usePopupChild = false;
        ImGui::Checkbox("Use popup child", &usePopupChild);

        if (popup)
        {
            ImGui::BeginPopup(&popup);

            if (usePopupChild)
                ImGui::BeginChild("popup child", ImVec2(100.0f, 100.0f));

            if (ImGui::Selectable("abcd"))
                popup = false;
            if (ImGui::IsItemHovered())
                ImGui::SetTooltip("abcd");

            if (ImGui::Selectable("efgh"))
                popup = false;
            if (ImGui::IsItemHovered())
                ImGui::SetTooltip("efgh");

            if (ImGui::Selectable("hijk"))
                popup = false;
            if (ImGui::IsItemHovered())
                ImGui::SetTooltip("hijk");

            if (usePopupChild)
                ImGui::EndChild();

            ImGui::EndPopup();
        }

        ImGui::End();
```

When you open the popup with a child region, the tooltips don't show up when hovering over the popup items until you click inside the region, then the tooltips will show.  Without the child region, the tooltips show up as expected.

The way I'm using the popups in my game, I use it to bring up a file selection interface.  Oftentimes, I'll have thousands of entries that need to be viewed, so I use the child region to keep the popup small and allow for scrolling:

![untitled](https://cloud.githubusercontent.com/assets/3121968/7173449/5cc31750-e3bd-11e4-8ef8-fafbd0d960ee.png)

With the absence of horizontal scrolling, we use tooltips a lot to show file paths in the case that they're too large for a window/child region, but not a huge issue by any means.

Side note: Would it be possible to allow for the popup to have a border, the same way a child region can have a border?  The idea is to let the popup's area be more obvious; our color scheme makes the popup a little hard to spot sometimes!

Thanks again, keep up the awesome work!

-Dale Kim

**Comment by ocornut** [comment]
- Looking into it..

- On the matter of borders.
Essentially BeginPopup() should have the same API as BeginChild() ? Without the identifier. There's no identifier because there can only be one popup on screen. However I haven't tested edge cases of trying to use different popups from frame to frame and there might be leaks of size/positional data involved. I haven't thought deeply if the lack of identifier for popups can eventually be a problem or not.

I was wondering if 'bool border' could be removed and be part of flags, 'ImGuiWindowFlags_Border', that depends how common border are. The current child parameter is odd side-stepping from the regular PushStyle/PopStyle system (yet somehow useful? or not? wouldn't it make more sense to bake it in the style?)

- Horizontal scrolling: yeah..probably not hard to add, need to see how it impacts (or not) auto-sizing features.

**Comment by ocornut** [comment]
OK so the bug isn't actually related to tooltip but a bug with IsItemHovered() and focused/child window. That also repros it:

```
        {
            ImGui::Begin("Popup tooltip bug");

            static bool popup = false;
            popup ^= ImGui::Button("Popup");

            static bool usePopupChild = false;
            ImGui::Checkbox("Use popup child", &usePopupChild);

            if (popup)
            {
                ImGui::BeginPopup(&popup);

                if (usePopupChild)
                    ImGui::BeginChild("popup child", ImVec2(100.0f, 100.0f));

                ImGui::Text("Hover me");
                if (ImGui::IsItemHovered())
                    ImGui::Text("Hovered");

                if (usePopupChild)
                    ImGui::EndChild();

                ImGui::EndPopup();
            }

            ImGui::End();
        }
```

**Comment by ocornut** [comment]
Main bug is fixed. I may just close that and add border options as a separate bug #

**Comment by Roflraging** [comment]
Tried it out and it works great now!  Definitely can close this if you want.

**Comment by ocornut** [comment]
Border: 

- I made popup have borders by default. It seemed unnecessary to add an option for now (if I do so I'd like to use the same parameters as BeginChild and would need to test the sizing options, but they seem unecessary for now). Let me know if you think there is a use for disabling the border.

- Their alpha is 1.0 (like combo box, not part of style yet).

- I also made border color use 65% alpha instead. Border color used for separator, etc. I think it looks better this way.

I don't want to worry too much about styling issue for now but the border is a readability issue. With it and the opaque window it is better.
